R/faasr_register_workflow_aws_lambda.R

```diff
@@ -0,0 +1,377 @@
+}
+
+# Get aws lambda function list
+faasr_register_workflow_lambda_function_lists <- function(faasr){
+    
+    function_list <- faasr$FunctionList
+    compute_servers <- faasr$ComputeServers
+
+    # initialize the list
+    lambda_function_info <- list()
+
+    # iterate over the functions
+    for (func_name in names(function_list)) {
+    
+        func <- function_list[[func_name]]
+        server_name <- func$FaaSServer
+        server_type <- compute_servers[[server_name]]$FaaSType
+        action_name <- func$Actionname
+
+        # check if the function server type is Lambda
+        if (server_type == "Lambda") {
+
+            if(check_lambda_exists(func_name)){
+                cat("\nlambda function -- ", func_name, "already exists.\n")
+                cat("Do you want to update it?[y/n]\n")
+                while(TRUE) {
+                    # check <- readLines(con = "stdin", 1)
+                    check <- readline()
+                    if(check == "y"){
+                    lambda_function_info[[func_name]]$action <- "update"
+                    break
+                    } else if(check == 'n'){
+                    cat("\nstop the script\n")
+                    stop()
+                    }else {
+                    cat("Enter \"y\" or \"n\": ")
+                    }
+                }
+            } else {
+                lambda_function_info[[func_name]]$action <- "create"
+            }
+            # add the function to the lambda function info list
+            lambda_function_info[[func_name]]$image <- action_name
+            
+        }
+    }
+    return (lambda_function_info)
+}
+
+# Get aws lambda function image list
+faasr_register_workflow_lambda_function_image <- function(faasr){
+    
+    function_list <- faasr$FunctionList
+    compute_servers <- faasr$ComputeServers
+
+    # initialize the lists
+    function_image_list <- list()
+
+    # iterate over the functions
+    for (func_name in names(function_list)) {
+        func <- function_list[[func_name]]
+        server_name <- func$FaaSServer
+        server_type <- compute_servers[[server_name]]$FaaSType
+        action_name <- func$Actionname
+        # check if the function server type is Lambda
+        if (server_type == "Lambda") {
+
+            # add the {action_name:image} pair to the function_image_list, if it's not present
+            if (!action_name %in% names(function_image_list)) {
+                # if "Actionname" is "fassr", then use faasr provided image--faasr/aws-lambda-tidyverse,
+                # otherwise use user provided image
+                # if(action_name == "faasr"){
+                #     image_path <- "faasr/aws-lambda-tidyverse"
+                # } 
+                action_name_value <- faasr$ActionContainers[[action_name]]
+                if(length(action_name_value)== 0 || action_name_value == ""){
+                    image_path <- "faasr/aws-lambda-tidyverse"
```

Comment:
- We don't have "latest" image in aws-lambda-tidyverse, so this line makes an error.
- This isn't working right now because we're in the stage of developing, but it should be working later.
`faasr/aws-lambda-tidyverse` -> `faasr/aws-lambda-tidyverse:0.0.0.5-dev`

Reply:
it's ok - we will have a "latest" tag as default, and as long as we can specify a particular tag in the JSON file, it's fine.